kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveRestoreConsumer.java

```diff
@@ -71,16 +123,18 @@ public void seek(final TopicPartition partition, final OffsetAndMetadata offsetA
             offsetAndMetadata.metadata()
         )
     );
+    uninitializedOffsets.remove(partition);
   }
 
   @Override
   public void seekToBeginning(final Collection<TopicPartition> partitions) {
-    final Collection<TopicPartition> withUncachedPosition = seekInitial(partitions);
+    final Set<TopicPartition> withUncachedPosition = initializeOffsets(partitions);
     if (withUncachedPosition.size() > 0) {
       // Make sure to only do this if the list of partitions without cached start positions
       // is not empty. If it's empty, KafkaConsumer will seek to the actual beginning of _all_
       // partitions.
       super.seekToBeginning(withUncachedPosition);
+      uninitializedOffsets.removeAll(withUncachedPosition);
```

Comment:
I think this call won't actually do anything, because `initializeOffsets` clears all the partitions it gets. Consider clearing just the ones with cached partitions from `initializeOffsets` or just clearing them all here (just for cleanliness sake - I don't think we're doing anything functionally wrong).